query/src/servers/http/middleware.rs

```diff
@@ -30,30 +30,46 @@ use poem::Middleware;
 use poem::Request;
 
 use super::v1::HttpQueryContext;
+use crate::servers::HttpHandlerKind;
 use crate::sessions::SessionManager;
 use crate::sessions::SessionType;
 use crate::users::auth::auth_mgr::Credential;
 
 pub struct HTTPSessionMiddleware {
+    pub kind: HttpHandlerKind,
     pub session_manager: Arc<SessionManager>,
 }
 
-fn get_credential(req: &Request) -> Result<Credential> {
+fn get_credential(req: &Request, kind: HttpHandlerKind) -> Result<Credential> {
     let auth_headers: Vec<_> = req.headers().get_all(AUTHORIZATION).iter().collect();
     if auth_headers.len() > 1 {
         let msg = &format!("Multiple {} headers detected", AUTHORIZATION);
         return Err(ErrorCode::AuthenticateFailure(msg));
     }
-    if auth_headers.is_empty() {
-        return Err(ErrorCode::AuthenticateFailure(
-            "No authorization header detected",
-        ));
-    }
     let client_ip = match req.remote_addr().0 {
         Addr::SocketAddr(addr) => Some(addr.ip().to_string()),
         Addr::Custom(..) => Some("127.0.0.1".to_string()),
         _ => None,
     };
+    if auth_headers.is_empty() {
+        if let HttpHandlerKind::Clickhouse = kind {
+            let (user, key) = (
+                req.headers().get("X-CLICKHOUSE-USER"),
```

Comment:
Is the naming used by clickhouse? How about using `X-CLICKHOUSE-USERNAME` and `X-CLICKHOUSE-PASSWORD` to match with HTTP basic auth?

Reply:
clickhouse just use it, we need to be compatible